src/raft_server.cxx

```diff
@@ -409,6 +409,12 @@ void raft_server::apply_and_log_current_params() {
           params->use_bg_thread_for_snapshot_io_ ? "ASYNC" : "BLOCKING",
           params->parallel_log_appending_ ? "ON" : "OFF" );
 
+    if (params->election_timeout_lower_bound_ >= params->election_timeout_upper_bound_) {
+        p_ft("Params election_timeout_upper_bound_ should larger than "
+             "election_timeout_lower_bound_");
+        ::exit(-1);
```

Comment:
Given that parameters can be changed at any time on the fly, it should be handled smoothly rather than killing the process. I'd recommend automatic adjustment of the upper bound:
```c++
if (params->election_timeout_lower_bound_ >= params->election_timeout_upper_bound_) {
    params->election_timeout_upper_bound_ = params->election_timeout_lower_bound_ * 2;
    p_in("invalid upper bound detected, adjusted to XX", ...);
}
```
Note that the same thing should be done for `heartbeat` and `lower_bound` as well (`lower_bound` should be greater than `heartbeat`).

If you really want to kill the process in your case, you can add a callback type (let's say `InvalidParameterDetected`) to notify such invalid parameters:
https://github.com/eBay/NuRaft/blob/1c2587f159775a4924573db64ffac5c973230761/include/libnuraft/callback.hxx#L28
and add
```c++
ctx_->cb_func_.call(cb_func::InvalidParameterDetected, &param);
```
in your callback handler, you can do `exit(-1)`.

Reply:
Thanks for advice.